src/ConsoleContactAppManager/ProgramHelpers.cs

```diff
@@ -0,0 +1,225 @@
+                Console.WriteLine($"Notes:{item.Notes}");
+                Console.WriteLine(" ");
+            }
+        }
+
+        /// <summary>
+        /// It searches the Contact List.
+        /// </summary>
+        /// <param name="contactList">List of stored contacts.</param>
+        public static void SearchContact(List<Contact> contactList)
+        {
+            // To check if there are no contacts in the list.
+            if (!contactList.Any())
+            {
+                Console.WriteLine("No Contacts to display");
+                return;
+            }
+
+            string nameToSearch;
+            List<Contact> searchcontactList = new ();
+            Console.WriteLine("Enter the name to search: ");
+            nameToSearch = Validators.GetStringInput();
+            foreach (Contact contact in contactList)
+            {
+                if (contact.Name is not null) // To change null state to non-null state before dereferencing.
+                {
+                    if (contact.Name.Contains(nameToSearch))
+                    {
+                        searchcontactList.Add(contact);
+                    }
+                }
+            }
+
+            DisplayContact(searchcontactList);
+        }
+
+        /// <summary>
+        /// Removes the contact that is to be deleted and updates the list
+        /// </summary>
+        /// <param name="contactList">updated list</param>
+        public static void DeleteContact(List<Contact> contactList)
+        {
+            if (!contactList.Any())
+            {
+                Console.WriteLine("No Contacts to display");
+                return;
+            }
+
+            bool flag = false;
+            DisplayContact(contactList);
+            Console.WriteLine("Enter the ID to be deleted:");
+            int idToRemove = Validators.GetInput();
+            if (idToRemove > 0 && idToRemove <= contactList.Count)
+            {
+                foreach (Contact contact in contactList.ToList())
+                {
+                    if (contact.Id == idToRemove)
+                    {
+                        contactList.RemoveAt(idToRemove - 1);
+                        flag = true;
+                    }
+
+                    // To update the contact id after deletion of a contact.
+                    else if (flag)
+                    {
+                        contact.Id--;
+                    }
+                }
+            }
+            else
+            {
+                Console.WriteLine("Enter valid id");
+            }
+
+            Console.WriteLine("Contact Removed.");
+        }
+
+        /// <summary>
+        /// EditContact method changes the list by user and updates the contact list.
+        /// </summary>
+        /// <param name="contactList">Takes the entire contact list.</param>
+        /// <returns>updated list</returns>
+        public static List<Contact> EditContact(List<Contact> contactList)
```

Comment:
You don't have to return the list every time as it is a reference.

Reply:
List is made global and appended for each method instead of returning a reference each time.